modules/options.nix

```diff
@@ -47,6 +63,15 @@ in {
     )
     optionsAttrs;
 
+  imports =
+    map
+    ({
+      from,
+      to,
+    }:
+      mkRenamedOptionModule [from] [to])
```

Comment:
Won't this create a warning for users that set `options` ? I'd rather have a silent rename

Reply:
Home-manager has a nice way to handle such things with a `stateVersion` that toggles warnings such that new users have deprecation warnings and old users don't have any warnings, maybe we should look into having something like that?

@GaetanLepage ?

Reply:
> Won't this create a warning?

Yes, [`mkAliasOptionModule`](https://github.com/NixOS/nixpkgs/blob/ce39273a07499c89d3fcb42989583098cea295f8/lib/modules.nix#L1225-L1231) would avoid the warning.

[`mkRenamedOptionModuleWith`](https://github.com/NixOS/nixpkgs/blob/ce39273a07499c89d3fcb42989583098cea295f8/lib/modules.nix#L1127-L1145) has a `sinceRelease` parameter, otherwise we'd have to introduce our own [`doRename`](https://github.com/NixOS/nixpkgs/blob/ce39273a07499c89d3fcb42989583098cea295f8/lib/modules.nix#L1259-L1355) wrapper.

Reply:
Hmmm not exactly sure, because sinceRelease would trigger for everyone, at a certain date. Due to our model with very few backports it is not _very_ useful.

I think we would need our own doRename wrapper indeed. But I don't want to decide unilateraly on this though

Reply:
Searching home-manager's codebase, [`stateVersion` only seems to be used](https://github.com/search?q=repo%3Anix-community%2Fhome-manager+stateVersion+language%3ANix&type=code&l=Nix&p=1) to configure default options.

Does nixvim even have a stateVersion (in standalone mode)?

I can also find [plenty of usage](https://github.com/search?q=repo%3Anix-community%2Fhome-manager+language%3ANix+renamed&type=code) of `mkRenamedOptionModule`, but none of `mkReneamedOptionModuleWith` or `doRename`.

Reply:
Well the thing I am trying to articulate is the following situations (but I'm not sure if 1. It is possible and 2. It is desirable):
- people that have been using the `options` option before this PR will _never_ have any warning
- (new) people that use the `options` option after this PR will _always_ have a warning

I think we don't have the correct infrastructure for this.

For an example of a home-manager option that warns differently depending on the HM stateversion: https://github.com/nix-community/home-manager/blob/c0ef0dab55611c676ad7539bf4e41b3ec6fa87d2/modules/programs/waybar.nix#L273-L282

Reply:
> - people that have been using the options option before this PR will never have any warning
> - (new) people that use the options option after this PR will always have a warning

I don't think this is possible, since we don't have a "nixvim" `stateVersion`.

Adding a state version would also imply to users that we'll maintain compatibility with old state versions indefinitely...

I believe we have to choose between either everyone or no-one getting a warning.

Reply:
Yeah I think too. Maybe we should add a stateVersion at some point to be able to do stuff like this

Between the no one and everyone I prefer the everyone (as this PR does)

Reply:
I'll wait for @GaetanLepage's thoughts before marking the thread resolved

Reply:
I'm ok with that too.